src/array/ops/cast.rs

```diff
@@ -431,6 +483,116 @@ fn extract_python_like_to_list<
     )
 }
 
+#[cfg(feature = "python")]
+fn extract_python_like_to_image_struct<
+    Tgt: numpy::Element + NumCast + ToPrimitive + arrow2::types::NativeType,
+>(
+    py: Python<'_>,
+    python_objects: &PythonArray,
+    child_dtype: DataType,
+    mode_from_dtype: Option<ImageMode>,
+) -> DaftResult<StructArray> {
+    // 3 dimensions - channel x height x width.
+
+    let shape_size = 3;
+    let (values_vec, offsets, shapes) =
+        extract_python_to_vec::<Tgt>(py, python_objects, &child_dtype, None, Some(shape_size))?;
+
+    let offsets = offsets.expect("Offsets should but non-None for image struct array");
+    let shapes = shapes.expect("Shapes should be non-None for image struct array");
+
+    let values_array: Box<dyn arrow2::array::Array> =
+        Box::new(arrow2::array::PrimitiveArray::from_vec(values_vec));
+
+    let inner_dtype = child_dtype.to_arrow()?;
+
+    let data_dtype = arrow2::datatypes::DataType::LargeList(Box::new(
+        arrow2::datatypes::Field::new("data", inner_dtype, true),
+    ));
+
+    let validity = python_objects.as_arrow().validity();
+
+    let data_array = Box::new(arrow2::array::ListArray::new(
+        data_dtype.clone(),
+        arrow2::offset::OffsetsBuffer::try_from(offsets)?,
+        values_array,
+        validity.cloned(),
+    ));
+
+    let mut channels = Vec::<u16>::with_capacity(shapes.len());
+    let mut heights = Vec::<u32>::with_capacity(shapes.len());
+    let mut widths = Vec::<u32>::with_capacity(shapes.len());
+    let mut modes = Vec::<u8>::with_capacity(shapes.len());
+    for shape in shapes.iter() {
```

Comment:
Ah good point, nulls current populate the shape vector with a dummy `(1,)`, which we don't handle well here. Agreed that validity should be taken into account here, I'll make that change.

I'll also add a bit more test coverage.

Reply:
Now that I think about it, the one place this could fail is if no type-level `mode` is given, and the dummy channel dim size of `1` is paired with a float32 dtype, which would cause the inferred mode to be invalid and fail.

Let me take a pass at pairing this shape vec iteration with the validity mask.